packages/firebase_admob/ios/Classes/FLTMobileAd.m

```diff
@@ -101,70 +101,108 @@ - (NSString *)description {
+  if (_anchorType == 0) {
+    verticalConstraint = [self.view.bottomAnchor constraintEqualToAnchor:layoutGuide.bottomAnchor
+                                                                constant:_anchorOffset];
   } else {
-    y = _anchorOffset;
+    verticalConstraint = [self.view.topAnchor constraintEqualToAnchor:layoutGuide.topAnchor
+                                                             constant:_anchorOffset];
   }
-  self.adView.frame = (CGRect){{x, self.adView.frame.origin.y}, self.adView.frame.size};
+
+  [NSLayoutConstraint activateConstraints:@[
+    verticalConstraint,
+    [self.view.centerXAnchor constraintEqualToAnchor:layoutGuide.centerXAnchor
+                                            constant:_horizontalCenterOffset],
+    [self.view.leftAnchor constraintGreaterThanOrEqualToAnchor:layoutGuide.leftAnchor],
+    [self.view.rightAnchor constraintLessThanOrEqualToAnchor:layoutGuide.rightAnchor],
+  ]];
+}
+
+- (void)activateConstraintParentView:(UIView *_Nonnull)parentView
+                   verticalAttribute:(NSLayoutAttribute)verticalAttribute {
+  [parentView
+      addConstraint:[NSLayoutConstraint constraintWithItem:self.view
+                                                 attribute:NSLayoutAttributeLeft
+                                                 relatedBy:NSLayoutRelationGreaterThanOrEqual
+                                                    toItem:parentView
+                                                 attribute:NSLayoutAttributeLeft
+                                                multiplier:1
+                                                  constant:0]];
+  [parentView addConstraint:[NSLayoutConstraint constraintWithItem:self.view
+                                                         attribute:NSLayoutAttributeRight
+                                                         relatedBy:NSLayoutRelationLessThanOrEqual
+                                                            toItem:parentView
+                                                         attribute:NSLayoutAttributeRight
+                                                        multiplier:1
+                                                          constant:0]];
+  [parentView addConstraint:[NSLayoutConstraint constraintWithItem:self.view
+                                                         attribute:NSLayoutAttributeCenterX
+                                                         relatedBy:NSLayoutRelationEqual
+                                                            toItem:parentView
+                                                         attribute:NSLayoutAttributeCenterX
+                                                        multiplier:1
+                                                          constant:_horizontalCenterOffset]];
+
+  [parentView addConstraint:[NSLayoutConstraint constraintWithItem:self.view
+                                                         attribute:verticalAttribute
```

Comment:
What about adding a comment that provides some context about the possible values that `verticalAttribute` can take?  Maybe add `NSAssert`.